### ISSUE TITLE
gaps in underline

### ISSUE THREAD

**Issue by rizen** [mimshwright/pixi-tagged-text#288]
```
<body fontFamily="Lato" fontStyle="Bold" underlineThickness="3" fontSize="42">Ability:  Laser Beam </body>
```
results in small gaps when spaces are present

<img width="383" alt="Screen Shot 2022-09-12 at 9 08 01 AM" src="https://user-images.githubusercontent.com/170009/189675582-b1c2bea0-0d84-4f32-8f8e-f2b2e1ebf8b5.png">

**Comment by rizen** [comment]
It happens with other fonts as well, not just Lato. Here it is with Arimo for example.

<img width="425" alt="Screen Shot 2022-09-12 at 9 11 12 AM" src="https://user-images.githubusercontent.com/170009/189676206-20c3245a-99bf-4b67-a985-a748732099a7.png">

**Comment by mimshwright** [comment]
The underline is drawn separately for each "word" and other symbols that are independent segments of the line of text, like spaces. So some subpixel difference there is probably causing this. I'm a little rusty on how the underline works when each letter is separate but my guess is you would see the gaps between each letter.

I have a couple of ideas of how to fix this...
*1. Add some kind of padding of x pixels on either side of each word for the underline.* This is probably a pretty quick fix. The overdraw amount could be variable.
*2. Try to refactor it to add only a single underline that stretches across the whole thing.* This one could be more performant since there's less drawing but probably not too noticeable. Plus it would probably break if you try to move the words around independently as in the animation in the demo

So I guess 1. is probably preferable.

**Comment by rizen** [comment]
1 sounds right

**Comment by mimshwright** [comment]
Fixed in v3.9.0
I've added `overdrawDecorations` as an option for TaggedText. This is a pixel value that is added to _both_ sides of the decoration. Default value is 0. It doesn't affect the layout of the text at all, only the size of the underlines. Note, the value is applied to all underlines in the whole text field so there is no individual control on a per-style basis.

**Comment by mimshwright** [comment]
# Before
![image](https://user-images.githubusercontent.com/141928/204795503-b3117dd9-aeb3-4200-acbc-f91d2d12d468.png)

# After (+10px)
![image](https://user-images.githubusercontent.com/141928/204795681-b6e6e122-d815-478d-a30e-7e213cc8540e.png)
